pytng/pytng.pyx

```diff
@@ -148,30 +203,55 @@ cdef class TNGFile:
         fname_bytes = fname.encode('UTF-8')
         ok = tng_util_trajectory_open(fname_bytes, _mode, & self._traj)
         if ok != TNG_SUCCESS:
-            raise IOError("An error ocurred opening the file. {}".format(status_error_message[ok]))
+            raise IOError("An error ocurred opening the file. {}".format(
+                status_error_message[ok]))
 
         if self.mode == 'r':
-            ok = tng_num_frames_get(self._traj, &self._n_frames)
+            ok = tng_num_frames_get(self._traj, & self._n_frames)
             if ok != TNG_SUCCESS:
-                raise IOError("An error ocurred reading n_frames. {}".format(status_error_message[ok]))
+                raise IOError("An error ocurred reading n_frames. {}".format(
+                    status_error_message[ok]))
 
             ok = tng_num_particles_get(self._traj, & self._n_atoms)
             if ok != TNG_SUCCESS:
-                raise IOError("An error ocurred reading n_atoms. {}".format(status_error_message[ok]))
+                raise IOError("An error ocurred reading n_atoms. {}".format(
+                    status_error_message[ok]))
 
-            ok = tng_distance_unit_exponential_get(self._traj, &exponent)
+            ok = tng_distance_unit_exponential_get(self._traj, & exponent)
             if ok != TNG_SUCCESS:
-                raise IOError("An error ocurred reading distance unit exponent. {}".format(status_error_message[ok]))
+                raise IOError("An error ocurred reading distance unit exponent. {}".format(
+                    status_error_message[ok]))
             self.distance_scale = 10.0**(exponent+9)
 
         self.is_open = True
         self.step = 0
         self.reached_eof = False
 
+    def get_strides(self):
```

Comment:
@richardjgowers this finds the stride lengths in pos force and vel, so that then we can compute the actual number of frames as `nframes%stride_length` and not try and read when theres no data.

However, this isn't a scalable option as there are loads of possible blocks with their own stride lengths and we can't keep track of all of them. Spooling them off one by one at the C level with `tng_util_trajectory_next_frame_present_data_blocks_find()` is probably the best option.